AppSearchSample/app/build.gradle

```diff
@@ -41,35 +42,33 @@ android {
         }
     }
     compileOptions {
-        sourceCompatibility JavaVersion.VERSION_1_8
-        targetCompatibility JavaVersion.VERSION_1_8
+        sourceCompatibility JavaVersion.VERSION_17
```

Comment:
This (and minsdk) seems like it will significantly restrict what phones this sample will run on. Since this is a sample we intend for other people, can we make it run more broadly?